lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2020,18 +2051,31 @@ formGraphOp(SILTensorOpInfo &opInfo,
       if (constValue.getKind() != SymbolicValue::Integer)
         return diagnoseInvalidAttr("requires a constant integer");
       break;
-    case SILTensorOpInfo::OperandClass::Shape:
     case SILTensorOpInfo::OperandClass::Array:
       if (constValue.getKind() != SymbolicValue::Array)
         return diagnoseInvalidAttr("requires an array");
-      // TODO: Check that the elements are the right types.  We should reject
-      // a shape with floats (or other junk) in the elements.
       break;
-
-    case SILTensorOpInfo::OperandClass::ShapeArray:
+    case SILTensorOpInfo::OperandClass::Shape:
+      if (verifyShapeAttr(constValue))
+        return; // error already emitted.
+      break;
+    case SILTensorOpInfo::OperandClass::ShapeArray: {
+      if (constValue.getKind() != SymbolicValue::Array)
+        return diagnoseInvalidAttr("requires an array");
+      CanType eltType;
+      auto shapes = constValue.getArrayValue(eltType);
+      if (eltType->getString() != "TensorShape")
+        return diagnoseInvalidAttr("requires an array of TensorShape objects");
+      for (auto shape : shapes) {
+        if (verifyShapeAttr(shape))
+          return; // error already emitted.
+      }
+      break;
+    }
     case SILTensorOpInfo::OperandClass::ArrayElement:
       // Match logic from checkAndDiagnoseOperands.
-      assert(0 && "FIXME: array constant exprs aren't handled yet");
+      inst->dump();
+      assert(0 && "FIXME: array elem exprs aren't handled yet");
```

Comment:
This isn't reachable, please remove this.

Reply:
The latest code in the repo no longer has this issue